model/integration_action.go

```diff
@@ -530,3 +532,142 @@ func DoPostActionRequestFromJson(data io.Reader) *DoPostActionRequest {
 	json.NewDecoder(data).Decode(&o)
 	return o
 }
+
+func (r *OpenDialogRequest) isValid() error {
+
+	if len(r.URL) == 0 || !IsValidHttpUrl(r.URL) {
+		return errors.New("Invalid URL")
+	}
+
+	if len(r.TriggerId) == 0 {
+		return errors.New("Empty trigger id")
+	}
+
+	err := r.Dialog.isValid()
+
+	return err
+}
+
+func (d *Dialog) isValid() error {
+
+	if d.Title == "" || len(d.Title) > 24 {
+		return errors.New("invalid dialog title")
+	}
+
+	if d.IconURL != "" && !IsValidHttpUrl(d.IconURL) {
+		return errors.New("invalid icon url")
+	}
+
+	if d.IntroductionText == "" {
+		return errors.New("invalid introduction text")
+	}
+
+	if len(d.Elements) > 5 {
+		return errors.New("invalid number of dialog elements")
+	}
+
+	if reflect.TypeOf(d.NotifyOnCancel).Kind() != reflect.Bool {
+		return errors.New("invalid type notify_on_cancel")
+	}
+
+	if len(d.Elements) != 0 {
+		elementMap := make(map[string]bool)
+
+		for i := 0; i < len(d.Elements); i++ {
+			if elementMap[d.Elements[i].DisplayName] == true {
+				return errors.New("duplicate dialog element name")
+			}
+			elementMap[d.Elements[i].DisplayName] = true
+			dialogElementErr := d.Elements[i].isValid()
```

Comment:
Nit pick: separate the two code sections

```suggestion
			elementMap[d.Elements[i].DisplayName] = true
			
			dialogElementErr := d.Elements[i].isValid()
```